pkgs/characters/lib/src/grapheme_clusters/breaks.dart

```diff
@@ -330,52 +467,250 @@ int previousBreak(String text, int start, int end, int index) {
 
 /// The next break no earlier than [index] in `string.substring(start, end)`.
 ///
-/// The index need not be at a grapheme cluster boundary.
+/// Uniquely to this function, the index need not be at a grapheme cluster
+/// boundary. That means there may be need for look-behind to find a character
+/// where the exact state is known.
 int nextBreak(String text, int start, int end, int index) {
```

Comment:
[nit] This method is going from <50 lines to >150 lines. Consider whether any pieces  can be pulled out into composed methods.

Reply:
I'm considering whether it should be a state machine. It probably can be. Maybe part of the forwards automaton, with more states above the normal states, since that's the normal states it should end up in after figuring out what it needs from the look-behind.

Reply:
And I've done that. More tables, more better!